src/applications/login/components/MockAuthButton.jsx

```diff
@@ -1,24 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
 import environment from 'platform/utilities/environment';
-import ENVIRONMENT_CONFIGURATIONS from 'site/constants/environments-configs';
+import environments from 'site/constants/environments';
+import { mockLogin } from 'platform/user/authentication/utilities';
+import {
+  CSP_IDS,
+  SERVICE_PROVIDERS,
+} from 'platform/user/authentication/constants';
+import { Select } from '@department-of-veterans-affairs/component-library';
 
 export default function MockAuthButton() {
-  return (
-    <button
-      type="button"
-      aria-label="Mock Authentication"
-      className="usa-button mauth-button vads-u-margin-y--1p5 vads-u-padding-y--2"
-      onClick={() => {
-        const environ =
-          typeof process !== 'undefined'
-            ? process.env.BUILDTYPE
-            : environment.BUILDTYPE;
-        window.location = `${
-          ENVIRONMENT_CONFIGURATIONS[environ].API_URL
-        }/v0/sign_in/authorize?client_id=vamock`;
-      }}
-    >
-      Mock Authentication
-    </button>
+  const [authType, setAuthType] = useState(CSP_IDS.LOGIN_GOV);
+  const processTypeUndefined = typeof process === 'undefined';
+  const environ = !processTypeUndefined
+    ? process.env.BUILDTYPE
+    : environment.BUILDTYPE;
+  return environ === environments.LOCALHOST ||
+    environ === environments.VAGOVDEV ? (
+    <>
+      {processTypeUndefined && (
+        <Select
+          label="Authentication Type"
+          name="authType"
+          includeBlankOption={false}
+          value={{ value: authType }}
+          onValueChange={({ value }) => setAuthType(value)}
+          options={Object.values(SERVICE_PROVIDERS).map(provider => ({
```

Comment:
Can you remove VA Mock from SERVICE_PROVIDERS?

Reply:
Definitely! Just to clarify, should it be removed from the `SERVICE_PROVIDERS` constant, or filtered out before mapping on this line?

Reply:
Remove it from the SERVICE_PROVIDERS as it's already a clientId 👍